File: app/src/main/java/me/lukeforit/spaceofaday/ui/archive/items/ApodItem.java
package me.lukeforit.spaceofaday.ui.archive.items;

import android.support.annotation.NonNull;

import me.lukeforit.spaceofaday.R;
import me.lukeforit.spaceofaday.data.model.Apod;

public class ApodItem extends ArchiveItem {

    @NonNull
    private Apod apod;

    public ApodItem(OnArchiveItemClickListener listener, @NonNull Apod apod) {
        super(listener);
        this.apod = apod;
    }

    @NonNull
    public Apod getApod() {
        return apod;
    }

    public void setApod(@NonNull Apod apod) {
        this.apod = apod;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.list_item_apod;
    }
}

File: app/src/main/java/me/lukeforit/spaceofaday/ui/archive/items/ApodItem.kt
package me.lukeforit.spaceofaday.ui.archive.items

import me.lukeforit.spaceofaday.R
import me.lukeforit.spaceofaday.data.model.Apod

class ApodItem(listener: ArchiveItem.OnArchiveItemClickListener, var apod: Apod) : ArchiveItem(listener) {
    override val layoutRes: Int
        get() = R.layout.list_item_apod
}
